pkg/users/usersexec.go

```diff
@@ -0,0 +1,135 @@
+package users
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+
+	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+	"github.com/percona/percona-server-mysql-operator/pkg/clientcmd"
+	"github.com/percona/percona-server-mysql-operator/pkg/mysql"
+)
+
+type dbExecImpl struct {
+	client *clientcmd.Client
+	pod    *corev1.Pod
+	user   apiv1alpha1.SystemUser
+	pass   string
+	host   string
+}
+
+func NewManagerExec(pod *corev1.Pod, user apiv1alpha1.SystemUser, pass, host string) (Manager, error) {
+	c, err := clientcmd.NewClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &dbExecImpl{client: c, pod: pod, user: user, pass: pass, host: host}, nil
+}
+
+func (d *dbExecImpl) exec(stm string) error {
+
+	cmd := []string{"mysql", "--database", "performance_schema", fmt.Sprintf("-p%s", d.pass), "-u", string(d.user), "-h", d.host, "-e", stm}
+
+	var outb, errb bytes.Buffer
+	err := d.client.Exec(context.TODO(), d.pod, "mysql", cmd, nil, &outb, &errb, false)
+	if err != nil {
+		return errors.Wrapf(err, "run %s, stdout: %s, stderr: %s", cmd, outb.String(), errb.String())
+	}
+
+	if strings.Contains(errb.String(), "ERROR") {
+		return fmt.Errorf("sql error: %s", errb.String())
+	}
+
+	return nil
+}
+
+// UpdateUserPasswords updates user passwords but retains the current password using Dual Password feature of MySQL 8
+func (d *dbExecImpl) UpdateUserPasswords(users []mysql.User) error {
+	err := d.exec("START TRANSACTION")
```

Comment:
@egegunes @hors 

The way this method (as well as `DiscardOldPasswords`) is implemented right now is just to mimic the implementation of the Manger with using SQL driver I believe this is not a good (or even the correct) way to do it with using `exec`.

Here we are starting the transaction, doing some operation, and then committing, but since `d.exec` underneath basically performs different HTTP requests, we don't have a single session and therefore I don't think having a transaction like this is correct.

But I wouldn't want to update this in PR, this https://jira.percona.com/browse/K8SPS-205 task is already in progress and there is a great chance that the whole approach will change, so I will handle everything in this ticket.

Reply:
Well, even with previous version that transaction wasn't doing anything good. `ALTER USER` causes an implicit `COMMIT` before execution, [see docs](https://dev.mysql.com/doc/refman/8.0/en/implicit-commit.html). You can happily remove this code.